src/scripts/main.js

```diff
@@ -1,3 +1,40 @@
 'use strict';
 
-// write code here
+const bodyElement = document.body;
+
+const dataSalary = [...bodyElement.querySelectorAll('li[data-salary]')];
+const listPeople = [...bodyElement.querySelectorAll('li')];
+
+const salaries = dataSalary
+  .map((elem) => +elem.getAttribute('data-salary').replace(/[$,]+/g, ''))
+  .sort((a, b) => b - a);
+
+const sortedArray = [];
+
+function sortList(list) {
+  for (let i = 0; i < list.length; i++) {
+    const salary = salaries[i];
+    const stringSalary = salary
+      .toLocaleString('en-us')
+      .padStart(salary.toLocaleString('en-us').length + 1, '$');
+
+    const person = list.filter(
+      (elem) => elem.getAttribute('data-salary') === stringSalary,
+    );
```

Comment:
The filter condition is incorrect. The `stringSalary` variable includes a dollar sign, but the `data-salary` attribute value does not. Consider converting the `data-salary` attribute value to a number and comparing it with the `salary` variable instead.